src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -50,6 +50,47 @@ typedef int (*ExecuteAssemblyFunction)(
             const char* managedAssemblyPath,
             unsigned int* exitCode);
 
+#if defined(__LINUX__)
+#define symlinkEntryPointExecutable "/proc/self/exe"
+#elif !defined(__APPLE__)
+#define symlinkEntryPointExecutable "/proc/curproc/exe"
+#endif
+
+bool GetEntrypointExecutableAbsoluePath(std::string& entrypointExecutable)
+{
+    bool result = false;
+    
+    entrypointExecutable.clear();
+
+    // Get path to the executable for the current process using
+    // platform specific means.
+#if defined(__LINUX__) || !defined(__APPLE__)
+    
+    // On non-Mac OS, return the symlink that will be resolved by GetAbsolutePath
+    // to fetch the entrypoint EXE absolute path, inclusive of filename.
+    entrypointExecutable.assign(symlinkEntryPointExecutable);
```

Comment:
Minor nit: consistency in casing of "entrypoint" throughout this function. Some places have the 'p' capitalized and others don't.